kroxylicious-multitenant/src/main/java/io/kroxylicious/proxy/filter/multitenant/MultiTenantTransformationFilter.java

```diff
@@ -101,7 +121,14 @@
         SyncGroupRequestFilter,
         LeaveGroupRequestFilter,
         HeartbeatRequestFilter,
-        DescribeGroupsRequestFilter, DescribeGroupsResponseFilter {
+        DescribeGroupsRequestFilter, DescribeGroupsResponseFilter,
```

Comment:
Yeah, it is getting a bit unwieldy.  Being able to use composition pattern in these sorts of use-cases would be desirable. 

On thing that might force the issue is the need to map client-ids.  With the way the API stands right now, we'd have to have two MultiTenant filters - on handling the specific requests/responses (MTTF we have now) and a new  implementation of `io.kroxylicious.proxy.filter.RequestFilter` to do the client re-write.   It would be horrid if the user had to include both in their filter lists in the kroxylicious config. I've been toying with the idea of a filter list API so a filter author could bundle filters that designed to work as one could be configured in one shot.

Reply:
I raised an issue #348 on this subject.